### PR TITLE
Engine notes for renderer, physics, and profiles

### PR DESCRIPTION
**Pull request by csauve**
Some followup from Jakey, and a few other engine-related tidbits I gathered.

Content changes
* Add init.txt to the arguments page, rename page
* Found a new field in `blam.sav`
* Add a stub page for the physics engine
* Add some notes about the renderer: more PC regressions and a section about shadow maps and lighting

Non-content changes
* Add a syntax highlighting language `inittxt` which is styled like haloscript console (each line is a command) but without the "halo(" prefix added in CSS.

### CODE REVIEW

File: src/content/h1/engine/physics/readme.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Physics engine
+stub: true
+---
+The physics engine of Halo 1 is an in-house implementation which simulates the movement of dynamic objects like [units][unit], [items][item], [projectiles][projectile], and [point_physics][] dynamics systems like [particle_system][] and [weather_particle_system][]. Unlike Halo 2 and later titles, Halo 1 does _not_ use any middleware like [Havok][].
+
+Different object interactions are simulated in different ways. For example, vehicles use [physics][] to drive themselves and collide with other vehicles, but their [model_collision_geometry][] is used in collisions with projectiles, bipeds, and particles. The [scenario_structure_bsp][] collision BSP is tested different for vehicles and projectiles than it is for players, making [phantom BSP][scenario_structure_bsp#phantom-bsp] more apparent.
```
Comment (gbMichelle) [inline src/content/h1/engine/physics/readme.md:7]: Vehicles also use their physics to interact with scenery collision geometry. Adding this note would avoid questions

File: src/content/h1/engine/renderer/readme.md [hunk 2]
```diff
@@ -24,11 +41,14 @@ The renderer needed to be adapted for the range of user hardware for the PC port
 * The [fog][] screen layers effect for simulated volumetric fog does not render at all.
 * HUD shield meters are missing their flash effect when drained.
 * Monochrome bitmaps and p8 bump map formats are unsupported.
+* Custom Edition skips rendering some effects when they are not loaded yet, such as the engine lens flares in a10's intro or initial bullet hole decals.
```
Comment (gbMichelle) [inline src/content/h1/engine/renderer/readme.md:44]: This is done to avoid stutters btw

File: src/content/h1/engine/renderer/readme.md [hunk 3]
```diff
@@ -24,11 +41,14 @@ The renderer needed to be adapted for the range of user hardware for the PC port
...
   * Ripple map mipmaps for [shader_transparent_water][shader_transparent_water#known-issues] are reversed, with smaller mipmaps being used at closer distances. MCC is not affected.
   * [shader_transparent_glass][] which use bump maps use the wrong tag for cube map reflections in Custom Edition. They should use the shader's referenced cube map, but instead use the rasterizer _vector normalization_ bitmap referenced by [globals][].
   * Transparent shaders do not sort properly behind [shader_transparent_glass][]; they will shift in front and behind each other. Part indexes allow draw order to be set manually for objects through the [gbxmodel][], but there is no such method for the [scenario_structure_bsp][]; the best you can do is flag the transparent shader to _draw before water_ since glass shaders seem to render in the same stage as water.
   * [shader_transparent_plasma][] (energy shields) [does not render correctly][shader_transparent_plasma#known-issues] on some hardware, and always incorrectly in MCC.
+  * Environment decals using the double multiply blend mode, like those on the floor of the mission a10, do not render.
```
Comment (gbMichelle) [inline src/content/h1/engine/renderer/readme.md:50]: double multiply, or just multiply? Should double check that

**Review by gbMichelle — APPROVED** [review]
Had a few comments, but mostly looks good

File: src/content/h1/engine/renderer/readme.md (hunk 2 above)
Comment (csauve) [reply src/content/h1/engine/renderer/readme.md:44]: If they were in predicted resources would this not be the case?

Comment (gbMichelle) [reply src/content/h1/engine/renderer/readme.md:44]: Afaik predicted resources don't always nest far enough

File: src/content/h1/engine/renderer/readme.md (hunk 3 above)
Comment (Lolibox) [reply src/content/h1/engine/renderer/readme.md:50]: Double. I think that specific a10 decal is the only environment decal that uses double.

Comment (gbMichelle) [reply src/content/h1/engine/renderer/readme.md:50]: Okay, thank you!

**Pull request merged**